File: 백준/Silver/2847. 게임을 만든 동준이/게임을 만든 동준이.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws NumberFormatException, IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        
        int[] scores = new int[N];
        
        int result = 0;
        for (int i = 0; i < N; i++) {
        	st = new StringTokenizer(br.readLine());
        	scores[i] = Integer.parseInt(st.nextToken());
		}
        
        for (int i = N-1; i > 0; i--) {
        	if (scores[i-1] >= scores[i]) {
        		result += scores[i-1] - scores[i] + 1;
        		scores[i-1] = scores[i] - 1;
			}
		}
        System.out.println(result);
        br.close();
    }
}

File: 백준/Silver/2847. 게임을 만든 동준이/게임을 만든 동준이.py
import sys
input = sys.stdin.readline

N = int(input())

Scores = [0 for _ in range(N)]
result = 0
for i in range(N) :
    Scores[i] = int(input())

if N != 1 :
    for i in range(1, N) :
        if Scores[N-i-1] >= Scores[N-i] :
            result += Scores[N - i - 1] - Scores[N - i] + 1
            Scores[N - i - 1] = Scores[N - i] - 1

print(result)